### PR TITLE
update Ch

### PR DESCRIPTION
**Pull request by ruslandoga**
### Changes

This PR updates Ch to fix socket leak discovered in https://github.com/plausible/ch/issues/152 and query string escaping from https://github.com/plausible/ch/pull/147

### Tests
- [x] This PR does not require tests

### Changelog
- [x] This PR does not make a user-facing change

### Documentation
- [x] This change does not need a documentation update

### Dark mode
- [x] This PR does not change the UI

### CODE REVIEW

File: mix.lock [hunk 1]
```diff
@@ -10,7 +10,7 @@
...
-  "ch": {:hex, :ch, "0.2.2", "71ad16063ca0d40dee0376b22abe8e0cb48140cad8681dca6c0558d6597c3fbd", [:mix], [{:db_connection, "~> 2.0", [hex: :db_connection, repo: "hexpm", optional: false]}, {:decimal, "~> 2.0", [hex: :decimal, repo: "hexpm", optional: false]}, {:ecto, "~> 3.5", [hex: :ecto, repo: "hexpm", optional: true]}, {:jason, "~> 1.0", [hex: :jason, repo: "hexpm", optional: false]}, {:mint, "~> 1.0", [hex: :mint, repo: "hexpm", optional: false]}], "hexpm", "72e7d5cfb0153b1b5b18244fc76c9b34ef82bba2b21038aefa29ff62ba1e22cf"},
+  "ch": {:hex, :ch, "0.2.3", "3d54ab30219be1e75b8e775a44b9002ba0bf969b9d1977c774b5536ed86f6b7c", [:mix], [{:db_connection, "~> 2.0", [hex: :db_connection, repo: "hexpm", optional: false]}, {:decimal, "~> 2.0", [hex: :decimal, repo: "hexpm", optional: false]}, {:ecto, "~> 3.5", [hex: :ecto, repo: "hexpm", optional: true]}, {:jason, "~> 1.0", [hex: :jason, repo: "hexpm", optional: false]}, {:mint, "~> 1.0", [hex: :mint, repo: "hexpm", optional: false]}], "hexpm", "5939b95de67e3d25bb322ad0a63ac44d9c4f4e06e43697926d88fd907a73b89d"},
```
Comment (ruslandoga) [inline mix.lock:13]: Hex dIff: https://diff.hex.pm/diff/ch/0.2.2..0.2.3

File: mix.lock [hunk 2]
```diff
@@ -10,7 +10,7 @@
...
-  "ch": {:hex, :ch, "0.2.2", "71ad16063ca0d40dee0376b22abe8e0cb48140cad8681dca6c0558d6597c3fbd", [:mix], [{:db_connection, "~> 2.0", [hex: :db_connection, repo: "hexpm", optional: false]}, {:decimal, "~> 2.0", [hex: :decimal, repo: "hexpm", optional: false]}, {:ecto, "~> 3.5", [hex: :ecto, repo: "hexpm", optional: true]}, {:jason, "~> 1.0", [hex: :jason, repo: "hexpm", optional: false]}, {:mint, "~> 1.0", [hex: :mint, repo: "hexpm", optional: false]}], "hexpm", "72e7d5cfb0153b1b5b18244fc76c9b34ef82bba2b21038aefa29ff62ba1e22cf"},
+  "ch": {:hex, :ch, "0.2.4", "d510fbb5542d009f7c5b00bb1ecab73307b6066d9fb9b220600257d462cba67f", [:mix], [{:db_connection, "~> 2.0", [hex: :db_connection, repo: "hexpm", optional: false]}, {:decimal, "~> 2.0", [hex: :decimal, repo: "hexpm", optional: false]}, {:ecto, "~> 3.5", [hex: :ecto, repo: "hexpm", optional: true]}, {:jason, "~> 1.0", [hex: :jason, repo: "hexpm", optional: false]}, {:mint, "~> 1.0", [hex: :mint, repo: "hexpm", optional: false]}], "hexpm", "8f065d15aaf912ae8da56c9ca5298fb2d1a09108d006de589bcf8c2b39a7e2bb"},
```
Comment (ruslandoga) [inline mix.lock:13]: and `v0.2.4` re-applies query string fix from https://github.com/plausible/ch/pull/147

Comment (ruslandoga) [reply mix.lock:13]: https://diff.hex.pm/diff/ch/0.2.2..0.2.4

**Review by aerosol — COMMENTED** [review]
Nice! That mistake is super familiar 😅

**Review by aerosol — APPROVED** [review]

**Pull request merged**